### PR TITLE
OCPBUGS-23259: [release-4.12] Update leaderelection config to allow retries

### PR DESCRIPTION
**Pull request by npinaeva**
cherry-pick of https://github.com/openshift/ovn-kubernetes/pull/1992

Conflicts:
	go-controller/cmd/ovnkube/ovnkube.go - remove crash handling
	go-controller/pkg/config/config.go - remove clusterManager

### CODE REVIEW

File: go-controller/pkg/config/config.go [hunk 1]
```diff
@@ -131,11 +131,14 @@ var (
...
 
+	// Set Leaderelection config values based on
+	// https://github.com/openshift/enhancements/blame/84e894ead7b188a1013556e0ba6973b8463995f1/CONVENTIONS.md#L183
+
 	// MasterHA holds master HA related config options.
-	MasterHA = MasterHAConfig{
-		ElectionLeaseDuration: 60,
-		ElectionRenewDeadline: 30,
-		ElectionRetryPeriod:   20,
+	MasterHA = HAConfig{
```
Comment (rbbratta) [inline go-controller/pkg/config/config.go:138]: pkg/config/config.go:138:13: undefined: HAConfig

Comment (rbbratta) [reply go-controller/pkg/config/config.go:138]: `MasterHAConfig`

**Review by trozet — APPROVED** [review]
/approve
/label backport-risk-assessed

**Pull request merged**